packages/generator/tests/conftest.py

```diff
@@ -1,11 +1,44 @@
 """Shared fixtures for ni-measurement-plugin-sdk-generator tests."""
 
 import pathlib
+import sys
+from typing import Generator
 
 import pytest
+from ni_measurement_plugin_sdk_service.discovery._support import _get_registration_json_file_path
+from ni_measurement_plugin_sdk_service.measurement.service import MeasurementService
+
+from tests.utilities.discovery_service_process import DiscoveryServiceProcess
+from tests.utilities.measurements import test_measurement
 
 
 @pytest.fixture
 def test_assets_directory() -> pathlib.Path:
     """Gets path to test_assets directory."""
     return pathlib.Path(__file__).parent / "test_assets"
+
+
+@pytest.fixture(scope="session")
+def discovery_service_process() -> Generator[DiscoveryServiceProcess, None, None]:
+    """Test fixture that creates discovery service process."""
+    if sys.platform != "win32":
+        pytest.skip(f"Platform {sys.platform} is not supported for discovery service tests.")
+
+    try:
+        registration_json_file_exists = _get_registration_json_file_path().exists()
+    except FileNotFoundError:  # registry key not found
+        registration_json_file_exists = False
+    if not registration_json_file_exists:
+        pytest.skip("Registration file not found. Ensure the Measurement Plug-In SDK is installed.")
+
+    with DiscoveryServiceProcess() as proc:
+        yield proc
+
+
+@pytest.fixture(scope="module")
```

Comment:
It's kind of weird for conftest to define module-scope fixtures. If the fixture can be shared between tests and is expensive to create, why not session scope? If it's important to reset the fixture between tests, why not function scope (the default)? 

You probably copied this from a fixture that was in one of the test files. In that situation, module scope makes more sense.

Reply:
ok